File: 169.MajorityElement/majority.cc
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> m;
        for (int i = 0; i < int(nums.size()); i++) {
            int count = 0;
            map<int,int>::iterator it = m.find(nums[i]);
            if (it != m.end()) {
                count = it->second;
            }
            m[nums[i]] = ++count;
            if (count > int(nums.size()) / 2) {
                return nums[i];
            }
        }
        return -1;
    }
};

File: 169.MajorityElement/majority.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        d = {}
        for n in nums:
            d[n] = d.get(n, 0) + 1
            if d[n] > len(nums) / 2:
                return n

        return 0
